### ISSUE TITLE
Pact Step by Step

### ISSUE THREAD

**Issue by fosho** [pact-foundation/pact-js#241]
I'm just a bit confused on the steps to take to publish and verify a report using pact and specifically a pact broker.

Correct me if I'm wrong but this is what I'm thinking;

1. Generate a report locally
2. Publish to pact broker
3. Verify report against provider
4. Publish verification results to pact broker

**Comment by mefellows** [comment]
Yes, basically, but `s/report/pact` - Pact tests produce pact files, which contain the interactions expected by the consumer, for a given provider. The concept of report doesn't make sense to me.

See the following links for more about workflows:

* https://github.com/pact-foundation/pact_broker/wiki/Set-up-checklist
* https://docs.google.com/document/u/1/d/e/2PACX-1vRf1kSDccImNipOOm1G-bjcSs-ifbZjf1v54K-dIcq8BLKeFPAAm_bf_p71UKqkRMIx30QWWL-kN8TI/pubhttps://docs.google.com/document/u/1/d/e/2PACX-1vRf1kSDccImNipOOm1G-bjcSs-ifbZjf1v54K-dIcq8BLKeFPAAm_bf_p71UKqkRMIx30QWWL-kN8TI/pub
* and of course: https://github.com/pact-foundation/pact_broker/

**Comment by fosho** [comment]
Thank you. 

So to clarify, the consumer test is used to generate a pact file based on provider data? And the provider test is used to verify that pact file?

The second link doesn't seem to be available by the way.

Thanks for all your help!

**Comment by abubics** [comment]
The second link was just pasted twice by accident. You should be able to c/p [half of it](https://docs.google.com/document/u/1/d/e/2PACX-1vRf1kSDccImNipOOm1G-bjcSs-ifbZjf1v54K-dIcq8BLKeFPAAm_bf_p71UKqkRMIx30QWWL-kN8TI/pub) ;)

To clarify your question,

- yes, the consumer test suite is used to generate the pact file, but
- the provider has a verify step, not strictly any tests (in the unit-testing sense of the consumer).

Some people prefer to write the provider verification as a suite of tests, but it feels like a lot of extra work to me.
